src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -93,6 +103,22 @@ public TaskService(TaskRepository taskRepository) {
                 });
     }
 
+    /**
+     * Updates a task if it already exists in DB.
+     * Otherwise, will not do anything.
+     * @param task Task to update.
+     * @return An {@link Optional<Task>} if task exists, {@link Optional#empty()} otherwise.
+     */
+    public Optional<Task> updateTask(Task task) {
+        final Optional<Task> knownTask = taskRepository
+                .findByChainTaskId(task.getChainTaskId());
+        if (knownTask.isPresent()) {
+            return Optional.of(taskRepository.save(task));
+        } else {
+            return Optional.empty();
+        }
```

Comment:
```suggestion
        return taskRepository
                .findByChainTaskId(task.getChainTaskId())
                .map(existingTask -> taskRepository.save(task));
```
or
```suggestion
        return taskRepository.existsByChainTaskId(task.getChainTaskId())
                ? Optional.of(taskRepository.save(task))
                : Optional.empty();
```

Reply:
With a `.map` and a method reference, it's much cleaner. Thanks!

Reply:
Done in https://github.com/iExecBlockchainComputing/iexec-core/pull/484/commits/3df6dcf53f4735b8dbaecea4f108532047c1d217.

Reply:
Thanks. But with the method reference, the task fetched from db will be saved, not the one passed as argument).

Reply:
Right, I had misread your comment. Thanks!
Fixed in https://github.com/iExecBlockchainComputing/iexec-core/pull/484/commits/01e1299635cd12945bdd1be95e0c537e68df68f3.